### ISSUE TITLE
Bug: [no-unused-vars] `varsIgnorePattern` does not disable the rule for function arguments

### ISSUE THREAD

**Issue by srmagura** [typescript-eslint/typescript-eslint#5620]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.8.2&sourceType=module&code=MYewdgzgLgBA+lAFgSwgSQgeQNYEMCeMAvDAOSkDcAUFQGYCuYwUy4MtAFHLgFwxj0AtgCMApgCcAlDADeVAL41OABklA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tidmUQAmtAG4BDaKgwBtcNhwB3cU0gAaWXODZ1cqGIkBJAOadoiAAqj8+RNGUZIAPQD6kbRAC+2gLqzP7oA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
const _thisIsOkay = '';

function f(_a: number) {
}

f(0)
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "@typescript-eslint/no-unused-vars": [
      "warn",
      {   
        "varsIgnorePattern": "^_"
      }
    ]
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

`_a` should NOT be reported as an unused variable.

### Actual Result

`_a` is reported as an unused variable.

### Versions

| package | version |
| -- | -- |
| `@typescript-eslint/eslint-plugin` | `5.36.2` |
| `@typescript-eslint/parser` | `5.36.2` |
| `TypeScript` | `4.8.2` |
| `ESLint` | `8.15.0` |
| `node` | `web` |

**Comment by srmagura** [comment]
Nevermind, found there is a separate `argsIgnorePattern` option. My bad...
